src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt

```diff
@@ -0,0 +1,267 @@
+import net.atos.zac.admin.model.HumanTaskParameters
+import net.atos.zac.admin.model.MailtemplateKoppeling
+import net.atos.zac.admin.model.UserEventListenerParameters
+import net.atos.zac.admin.model.ZaakafhandelParameters
+import net.atos.zac.admin.model.ZaakbeeindigParameter
+import net.atos.zac.admin.model.ZaakbeeindigReden
+import net.atos.zac.util.UriUtil
+import net.atos.zac.util.ValidationUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.net.URI
+import java.time.ZonedDateTime
+import java.util.Optional
+import java.util.UUID
+import java.util.logging.Logger
+
+@ApplicationScoped
+@Transactional
+@AllOpen
+@NoArgConstructor
+@Suppress("TooManyFunctions")
+open class ZaakafhandelParameterBeheerService @Inject constructor(
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private val entityManager: EntityManager,
+    private val ztcClientService: ZtcClientService,
+    private val zaakafhandelParameterService: ZaakafhandelParameterService
+) {
+    companion object {
+        private val LOG: Logger = Logger.getLogger(ZaakafhandelParameterBeheerService::class.java.name)
+    }
+
+    fun readZaakafhandelParameters(zaaktypeUUID: UUID): ZaakafhandelParameters {
+        ztcClientService.readCacheTime()
+        val builder = entityManager.criteriaBuilder
+        val query = builder.createQuery(
+            ZaakafhandelParameters::class.java
+        )
+        val root = query.from(
+            ZaakafhandelParameters::class.java
+        )
+        query.select(root).where(builder.equal(root.get<Any>(ZaakafhandelParameters.ZAAKTYPE_UUID), zaaktypeUUID))
+        val resultList = entityManager.createQuery(query).resultList
+        if (resultList.isNotEmpty()) {
+            return resultList.first()
+        } else {
+            return ZaakafhandelParameters().apply {
+                zaakTypeUUID = zaaktypeUUID
+            }
+        }
+    }
+
+    fun listZaakafhandelParameters(): List<ZaakafhandelParameters> {
+        val query = entityManager.criteriaBuilder.createQuery(ZaakafhandelParameters::class.java)
+        val root = query.from(ZaakafhandelParameters::class.java)
+        query.select(root).orderBy(entityManager.criteriaBuilder.desc(root.get<Any>("id")))
+        return entityManager.createQuery(query).resultList
+    }
+
+    fun createZaakafhandelParameters(zaakafhandelParameters: ZaakafhandelParameters): ZaakafhandelParameters {
+        zaakafhandelParameterService.clearListCache()
+        ValidationUtil.valideerObject(zaakafhandelParameters)
+        zaakafhandelParameters.humanTaskParametersCollection.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.userEventListenerParametersCollection.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.mailtemplateKoppelingen.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.creatiedatum = ZonedDateTime.now()
```

Comment:
We can probably use `let`, `with` here

Reply:
I simplified this but in a different way. ;-)